servers/rendering/renderer_rd/storage_rd/light_storage.cpp

```diff
@@ -1718,6 +1718,33 @@ void LightStorage::set_max_reflection_probes(const uint32_t p_max_reflection_pro
 	reflection_buffer = RD::get_singleton()->storage_buffer_create(sizeof(ReflectionData) * max_reflections);
 }
 
+void LightStorage::sort_mobile_reflection_probes(const RID *p_reflection_probe_instances, RendererRD::ForwardID *p_reflection_probes, uint32_t reflection_probe_count) {
```

Comment:
I think this whole function can be simplified because you have already sorted the reflection probes by size when filling the reflection probe buffer. Therefore the index stored by the ForwardID is enough to sort them by size using `forward_id_allocators[p_type].map[p_id]`. You just have to ensure that they are stored by smallest to largest index. The nice benefit of doing that is you can keep all your logic inside `fill_push_constant_instance_indices()` and you don't have to pass around references to the instances either.

Reply:
This makes a ton of sense, the implementation is a lot cleaner now. :)